lib/mix/test/mix/tasks/archive_test.exs

```diff
@@ -149,6 +149,9 @@ defmodule Mix.Tasks.ArchiveTest do
       true = Application.compile_env!(:git_repo, :archive_config)
 
       defmodule GitRepo.Archive do
+        @app :git_repo
+        true = Application.compile_env!(@app, :archive_config)
```

Comment:
Can you please add a test to `test/elixir/application_test.exs` instead? This test is not really related to the functionality.